### ISSUE TITLE
Paper value import when pipe-separated entry in GPAD

### ISSUE THREAD

**Issue by vanaukenk** [geneontology/gocamgen#31]
From evidence rules 2 and 3 in the [Google doc](https://docs.google.com/document/d/1AgkJqA4-V0HG0SaDiNO9ENTxaQC_1HwsmwyhyHhuxog/edit#), we said that if there was more than one reference separated by a pipe, we would import only one reference using this descending list of preferences:

Order of preference for references:
PMID
GO_REF
doi
Other resource, e.g. MOD paper ID

Do we still want to do this?  

Right now, it looks like both IDs (i.e. PMID and WBPaper ID) are being imported from the WB annotations, but only the WBPaper ID is displayed in the graph editor and both are exported in the Annotation Preview.

I can't quite make out how the paper ID is selected for the corresponding form view; both seem to be used but I can't discern a pattern.

**Comment by vanaukenk** [comment]
Hi @dustine32 

I'm re-opening this ticket because I think there may be a case-sensitivity issue with the references that use a DOI rather than a PMID.

The abbreviation in the GPAD is actually 'DOI' rather than 'doi'.

This model has two examples of where the DOI is not being included as the Source, probably because 'DOI' is upper case:
http://noctua-dev.berkeleybop.org/editor/graph/gomodel:c2304f62-ce53-4ef9-9a3e-feeba6fcbe8f

At some point we will need to talk about what to display for a DOI in Noctua, but for now I think it's sufficient to just display the full string.

**Comment by dustine32** [comment]
Hey @vanaukenk , got this DOI showing up now in test model for [WB:WBGene00015350](http://68.181.46.18:8910/editor/graph/gomodel:9526f428-5295-43d6-a7b3-b3ce47f26e02):
![image](https://user-images.githubusercontent.com/2678599/58671576-02ae7a00-82f8-11e9-8952-a25c88ab81d1.png)

I just normalized the cases of the comparison strings.

**Comment by vanaukenk** [comment]
This looks good to me @dustine32 
Thanks!

**Comment by vanaukenk** [comment]
Update: 2021-03-02
Import references exactly as they are found in the incoming GPAD, so, for example, there may be lines with both a PMID and a MOD reference - both would be taken and pipe-separated.

**Comment by dustine32** [comment]
Update from 2021-05-25 MOD imports call:

Resurrect the reference preference code! Pick only one reference per annotation, preferring in this order (same as [top comment](https://github.com/geneontology/gocamgen/issues/31#issue-410437559) but clarified here):
1. PMID
2. GO_REF
3. doi
4. Anything else

Confirming with @vanaukenk @ukemi @sabrinatoro

**Comment by ukemi** [comment]
PMID
    GO_REF
    doi
    MOD reference
    Anything else?

**Comment by dustine32** [comment]
@ukemi "MOD reference" would just be a list of all MOD reference prefixes? Is a comprehensive list available somewhere or should I just handle MGI, WB, ZFIN, SGD references for now?

Also, is there an MGI case where the reference field would contain something like `MGI:MGI:2154458|ANYTHING_ELSE:1234` that I can use to test?

**Comment by ukemi** [comment]
Hi @dustine32   I don't think we would have that example at MGI. They would all contain an MGI:MGI: value piped with one of the other types in the list.  

As for the prefixes, maybe in a YAML file somewhere, but that assumes that all databases use the same prefix for all database objects. This is a good point I hadn't considered. Maybe stick to your original plan?

**Comment by dustine32** [comment]
@ukemi Right, I agree we probably don't need to consider all MOD references prefixes right now. We can always implement some more nuanced logic later if needed though.
